abstract/group.go

```diff
@@ -51,6 +51,9 @@ type Secret interface {
 
 	// Set to a fresh random or pseudo-random secret
 	Pick(rand cipher.Stream) Secret
+
+	// SetBytes will take bytes and create a secret out of it
+	SetBytes([]byte) Secret
```

Comment:
Oooh - I'm pretty sure you really don't want to depend on being able to do a == a2 even in the MarshalBinary case!  Is there code that is doing that?  Sounds like a really bad idea to me - whether it works would definitely depend on which particular implementation of Secret or Point we're looking at.  For Point.MarshalBinary(), for example, I think special encoding is always going to be required, which will always allocate a new byte-slice.  If that's not happening for ed25519.Secret, then that might be a bug that needs to be fixed.